tests/stdlib/tsugar.nim

```diff
@@ -1,4 +1,72 @@
-import sugar
+discard """
+  output: '''
+x + y = 30
+'''
+"""
+
+import std/sugar
+
+# `=>`
+
+block:
+  let f = () => 42
+  doAssert f() == 42
+
+  let g = (x: int) => x + 1
+  doAssert g(42) == 43
+
+  let h = (x, y: int) => x + y
+  doAssert h(1, 2) == 3
+
```

Comment:
nit: i usually prefer f1, f2 etc insetead of f, g when there's a lot of such procs to test
```suggestion
  let f2 = () => ()
  doAssert f2() == () # empty tuple, not void

  const f3 = () => ()
  doAssert f3() == ()

  # const f4 = () => void # xxx we should have a way to generate a void-returning proc 
```

Reply:
What's the point of testing `() => ()`?

Also, not sure what you mean by the last comment, returning `void` seems to work.

Reply:
indeed, following up in https://github.com/nim-lang/Nim/pull/16820#discussion_r567195878